### PR TITLE
Make the tests pass

### PR DESCRIPTION
**Pull request by jakzal**
This is by no means a finished implementation. It merely demonstrates how we can use "compose" functions instead of "apply" ones, to fix the problem with nested completes. 

Apply is to compose as map is to flatMap. Compose will flatten the result of composed CompletableFuture(s).

### CODE REVIEW

File: src/main/java/io/vlingo/common/completes/CFCompletes.java [hunk 1]
```diff
@@ -751,6 +751,16 @@ public String toString() {
       };
     }
 
+    private <O> Function<T, CompletableFuture<O>> composableFunction(final Function<T, O> userFunction) {
+      return (T t) -> {
+        O outcome = userFunction.apply(t);
+        if (outcome instanceof CFCompletes) {
+          return (CompletableFuture<O>) ((CFCompletes<?>) outcome).state().future;
```
Comment (jakzal) [inline src/main/java/io/vlingo/common/completes/CFCompletes.java:758]: Unfortunately, I'm relying on a specific implementation of Completes here. I haven't found a better way to do this (yet).

Comment (VaughnVernon) [reply src/main/java/io/vlingo/common/completes/CFCompletes.java:758]: @jakzal What do you need that's different? Does the Completes interface API need to change? Feel free to make changes, hopefully non-breaking ones.

Comment (jakzal) [reply src/main/java/io/vlingo/common/completes/CFCompletes.java:758]: Is it possible we'd be dealing with two different Completes implementations here? i.e. CFCompletes and SomeCustomCompletes.

Comment (VaughnVernon) [reply src/main/java/io/vlingo/common/completes/CFCompletes.java:758]: Nope. `BlahBlahCompletes<T>`, where `BlahBlah` is currently `CF`, is it.

**Comment by jakzal** [comment]
replaced by https://github.com/vlingo/vlingo-common/pull/34

**Pull request closed without merging**